front/ui/ui-charts/src/manchette/hooks/useManchetteWithSpaceTimeChart.tsx

```diff
@@ -118,53 +115,49 @@ const useManchetteWithSpaceTimeChart = ({
   splitPoints?: SplitPoint[];
   options?: Partial<ManchetteWithSpaceTimeChartOptions>;
 }) => {
+  const canvasDrawingHeight = Math.max(1 + BASE_WAYPOINT_HEIGHT, height - FOOTER_HEIGHT); // 521
+  const drawingHeightWithoutTopPadding = canvasDrawingHeight - verticalPadding; // 473
+  const drawingHeightWithoutBothPadding =
+    canvasDrawingHeight - (verticalPadding + BASE_WAYPOINT_HEIGHT); // 441
+
   const { displayTimeCaptions, enableTimePan, enableSpacePan, enableTimeZoom } = useMemo(
     () => ({
       ...DEFAULT_MANCHETTE_WITH_SPACE_TIME_CHART_OPTIONS,
       ...options,
     }),
     [options]
   );
-  const [isShiftPressed, setIsShiftPressed] = useState(false);
-  const [state, setState] = useState<State>({
-    xZoom: timeScaleToZoomValue(DEFAULT_ZOOM_MS_PER_PX),
-    yZoom: 1,
-    timeOrigin: defaultTimeOrigin,
-    spaceOrigin: defaultSpaceOrigin,
-    xOffset: defaultXOffset,
-    yOffset: 0,
-    scrollTo: null,
-    panning: null,
-    zoomMode: false,
-    rect: null,
-    pixelRect: null,
-    isProportional: true,
-    waypointsChart: [],
-    scales: [],
-  });
+
+  const [isProportional, setIsPropotional] = useState(true);
 
   const {
-    xZoom,
-    yZoom,
-    timeOrigin,
-    spaceOrigin,
-    xOffset,
-    yOffset,
-    scrollTo,
-    panning,
-    zoomMode,
-    rect,
-    isProportional,
-  } = state;
+    syncManchetteState: { xZoom, yZoom, timeOrigin, spaceOrigin, xOffset, yOffset, zoomMode, rect },
+    setSyncManchetteState,
+    isShiftPressed,
+    handleScrollInManchette,
+    handleXZoom,
+    yZoomHelpers: { zoomYIn, zoomYOut, resetZoom },
+  } = useSyncManchette({
+    manchetteWithSpaceTimeChartRef,
+    diagramRef: spaceTimeChartRef,
+    height,
+    defaultTimeOrigin,
+    defaultSpaceOrigin,
+    defaultXOffset,
+    maxYZoom: isProportional
+      ? MAX_ZOOM_Y
+      : (drawingHeightWithoutTopPadding - BASE_WAYPOINT_HEIGHT) / (2 * BASE_WAYPOINT_HEIGHT),
+    options: { enableTimeZoom },
```

Comment:
I wonder if we should have separate hooks for X and Y, instead of having an option to disable half of the hook… Having a hook which only takes care of the Y axis will make it easier to use this hook in the standalone track occupancy diagram and crossing diagram.

But this can be done later.